backend/api/receipts/download.py

```diff
@@ -0,0 +1,64 @@
+from django.http import FileResponse, Http404
+from backend.models import Receipt, ReceiptDownloadToken
+from django.shortcuts import get_object_or_404, render
+from django.urls import reverse
+from django.http import HttpResponse, JsonResponse
+
+
+def download_receipt(request, token):
+    """
+    Downloads a receipt file based on the provided token.
+
+    Args:
+        request (HttpRequest): The HTTP request object.
+        token (str): The token used to authenticate the download.
+
+    Returns:
+        HttpResponse: The HTTP response containing the downloaded receipt file.
+    """
+    try:
+        download_token = ReceiptDownloadToken.objects.get(token=token)
+    except ReceiptDownloadToken.DoesNotExist:
+        return HttpResponse("Download link has been used", status=404)
+
+    # if download_token.is_used():
+    #     return HttpResponse("Download link has been used", status=410)  # 410 Gone
+    if download_token.user != request.user:
+        return HttpResponse("Invalid user", status=403)  # 403 Forbidden
+
+    receipt = get_object_or_404(Receipt, id=download_token.file.id)
+
+    download_token.delete()
+
+    response = FileResponse(receipt.image)
```

Comment:
I love the idea of passing an Image through a response, that means we don't need Nginx to handle it. Only downside is that image processing through the UI could be bad performance wise, but so far I don't see too much of an issue!

Reply:
Thank you for your feedback! Based on the current implementation, my understanding is that the solution does not render the image in the UI; rather, it directly initiates the download of the image file when the user clicks the 'Download' button.